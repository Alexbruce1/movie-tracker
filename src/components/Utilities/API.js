const export fetchData = async (url) => {
	const response = await fetch(url)
	if (response.status >= 400) {
		
	} else {

	}
}