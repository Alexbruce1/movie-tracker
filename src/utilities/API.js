export const fetchData = async (url) => {
	const response = await fetch(url)
	if (response.status >= 400) {
		throw new Error('Error!')
	} else {
			const cleanResponse = await response.json()
			return cleanResponse
	}
}

export const loginUser = async (user) => {
  try {
    const response = await fetch('http://localhost:3000/api/users', {
      method: 'POST',
      body: JSON.stringify(user),
      headers:{
        'Content-Type': 'application/json'
      }
    })
    const userData = await response.json();
    return userData.data;
  }
  catch(error) {
    console.log(error.message);
  }
}

export const createUser = async (user) => {
  const userData = await getUsers()
  console.log(userData)
  const matchingUser = userData.find(userEntry => {
    return userEntry.email === user.email
  })
  if (!matchingUser) {
    fetch('http://localhost:3000/api/users/new', {
      method: 'POST',
      body: JSON.stringify(user),
      headers:{
        'Content-Type': 'application/json'
      }
    }); 
  } else {
    console.log('ERROR: User already exists')
    return null
  }
}

const getUsers = async () => {
  const response = await fetch('http://localhost:3000/api/users', {
    method: 'GET',
    body: JSON.stringify(),
    headers:{
      'Content-Type': 'application/json'
    }
  })
  const userData = await response.json();
  return userData.data
}







// export const fetchData = async (url) => {
//   const response = await fetch(url)
//   if (response.status >= 400) {
//     throw new Error('Error!')
//   } else {
//       const cleanResponse = await response.json()
//       return cleanResponse
//   }
// }

// export const loginUser = async (user) => {
//   try {
//     const response = await fetch('http://localhost:3000/api/users', {
//       method: 'POST',
//       body: JSON.stringify(user),
//       headers:{
//         'Content-Type': 'application/json'
//       }
//     })
//     const userData = await response.json();
//     return userData.data;
//   }
//   catch(error) {
//     console.log(error.message);
//   }
// }

// export const createUser = (user) => {
//   fetch('http://localhost:3000/api/users/new', {
//     method: 'POST',
//     body: JSON.stringify(user),
//     headers:{
//       'Content-Type': 'application/json'
//     }
//   });
// }
