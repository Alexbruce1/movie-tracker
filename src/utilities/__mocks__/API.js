/* eslint-disable */

export const fetchData = jest.fn().mockImplementation(() => {
	return { results: [] };
});

export const loginUser = jest.fn().mockImplementationOnce(() => ({
	id: 31,
	loggedIn: true
})).mockImplementationOnce(() => ({
	id: 31,
	loggedIn: false
}));

export const getFavorites = jest.fn().mockImplementation(() => (
	[]
));

export const createUser = jest.fn(() => ({  
	id: 31,
	status: 400
})).mockImplementationOnce(() => ({
	id: 31,
	status: 200
})).mockImplementationOnce(() => ({
	id: 31,
	status: 400
}));

export const addFavorite = jest.fn();
export const removeFavorite = jest.fn();