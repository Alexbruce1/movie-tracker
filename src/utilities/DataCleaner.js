import * as API from './API'

export const fetchMovies = async () => {
	const apiKey = '26d5b93e45b773596adda2d2b99efa0f'
	const url = `https://api.themoviedb.org/3/movie/now_playing?api_key=${apiKey}&language=en-US&page=1`

	const movieData = await API.fetchData(url)
	const currentMovies = await returnMovieData(movieData.results)
	return currentMovies
}

export const returnMovieData = async (movies) => {
	const moviePromises = movies.map( async movie => {

		return {
			title: movie.title,
			poster: `https://image.tmdb.org/t/p/w500/${movie.poster_path}`,
			overview: movie.overview,
			releaseDate: formatReleaseDate(movie.release_date),
			rating: movie.vote_average
		}
	})
	return Promise.all(moviePromises)
}

const formatReleaseDate = (date) => {
	const month = date.slice(5,7)
	const day = date.slice(8,10)
	const year = date.slice(0,4)
	return `${month}/${day}/${year}`
}