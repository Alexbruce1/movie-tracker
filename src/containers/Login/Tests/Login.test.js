import React from 'react';
import { shallow } from 'enzyme';
import { Login }  from '../';
import * as API from '../../../utilities/API';
jest.mock('../../../utilities/API');

describe('Login', () => {
  let wrapper;
  let mockEvent;
  let mockUser;
  let mockCredentials;
  let mockFunc;

  beforeEach(() => {
  	mockFunc = jest.fn()
    mockUser = {id: 31, loggedIn: false}
    mockCredentials = {email: '12345@gmail.com', password: 12345}
    wrapper = shallow(<Login loginUser={jest.fn()} setFavorites={mockFunc}/>);
    mockEvent = { 
    	preventDefault: jest.fn(), 
    	target: {name: 'name', value: 'Alex'} 
    }
    wrapper.setState({create: true})
  });

  it('Should render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  describe('handleChange', () => {
	  it('Sets state upon invocation of handleChange', () => {
	  	wrapper.instance().handleChange(mockEvent)
	  	expect(wrapper.state().name).toEqual('Alex')
	  });

  })

  describe('submitLogin', () => {
	  it('Invokes loginUser if loginAttempt is successful', async () => {
      await wrapper.instance().submitLogin(mockEvent)
	  	expect(wrapper.instance().props.loginUser).toHaveBeenCalled()
	  });

	  it('Invokes setFavorites if loginAttempt is successful', async () => {
      const expected = []
      await wrapper.instance().submitLogin(mockEvent)	  	
      expect(wrapper.instance().props.setFavorites).toHaveBeenCalledWith(expected)
	  });

	  it('Sets error state if loginAttempt fails', async () => {
	  	const response = 'Email and Password did not match'
	  	await wrapper.instance().submitLogin(mockEvent)
	  	expect(wrapper.state().error).toEqual(response)
	  });
  })

  describe('toggleCreate', () => {
	  it('Sets create and error states if create state is false and fetch response succeeds', async () => {
      const expected = ''
      await wrapper.instance().toggleCreate(mockEvent)
      expect(wrapper.state().create).toBe(false)
      expect(wrapper.state().error).toBe(expected)
	  });

    it('Sets error state if create state is true and fetch response fails', async () => {
      const expected = 'Email has already been used'
      await wrapper.instance().toggleCreate(mockEvent)
      expect(wrapper.state().error).toBe(expected)     
    });

    it('Clears inputs if create state is true and fetch response fails', async () => {
      await wrapper.instance().toggleCreate(mockEvent)
      expect(wrapper.state().email).toEqual('')
      expect(wrapper.state().password).toEqual('')
      expect(wrapper.state().name).toEqual('')        
    });

  })

  describe('newUserResponse', () => {
	  it('Invokes API.createUser upon invocation of newUserResponse', async () => {
      await wrapper.instance().newUserResponse(mockEvent)	  	
      expect(API.createUser).toHaveBeenCalled()
	  });
  })

  describe('clearInputs', () => {
	  it('Clears inputs', () => {
	  	wrapper.setState({
	      email: 'test',
	      password: 'test',
	      name: 'test'        
	    })   
	    wrapper.instance().clearInputs()
	    expect(wrapper.state().email).toEqual('')
	    expect(wrapper.state().password).toEqual('')
	    expect(wrapper.state().name).toEqual('')
	  });
  })
});
