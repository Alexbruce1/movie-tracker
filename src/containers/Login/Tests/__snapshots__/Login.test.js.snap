// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    loginUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="Login-form"
      >
        <h1
          className="Login-header"
        >
           Movie Tracker 
        </h1>
        <h3>
           Login 
        </h3>
        <input
          name="email"
          onChange={[Function]}
          placeholder="Email address"
          type="text"
          value=""
        />
        <input
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <button
          className="login-button"
          onClick={[Function]}
          type="submit"
        >
          Login
        </button>
        <button
          className="create-user"
          onClick={[Function]}
        >
          Create User
        </button>
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="home-link"
          to="/"
        >
          Home
        </NavLink>
      </form>,
      "className": "Login",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="Login-header"
          >
             Movie Tracker 
          </h1>,
          <h3>
             Login 
          </h3>,
          null,
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email address"
            type="text"
            value=""
          />,
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />,
          <button
            className="login-button"
            onClick={[Function]}
            type="submit"
          >
            Login
          </button>,
          <button
            className="create-user"
            onClick={[Function]}
          >
            Create User
          </button>,
          null,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="home-link"
            to="/"
          >
            Home
          </NavLink>,
        ],
        "className": "Login-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Movie Tracker ",
            "className": "Login-header",
          },
          "ref": null,
          "rendered": " Movie Tracker ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Login ",
          },
          "ref": null,
          "rendered": " Login ",
          "type": "h3",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email address",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "className": "login-button",
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create User",
            "className": "create-user",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Create User",
          "type": "button",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Home",
            "className": "home-link",
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="Login-form"
        >
          <h1
            className="Login-header"
          >
             Movie Tracker 
          </h1>
          <h3>
             Login 
          </h3>
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email address"
            type="text"
            value=""
          />
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <button
            className="login-button"
            onClick={[Function]}
            type="submit"
          >
            Login
          </button>
          <button
            className="create-user"
            onClick={[Function]}
          >
            Create User
          </button>
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="home-link"
            to="/"
          >
            Home
          </NavLink>
        </form>,
        "className": "Login",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="Login-header"
            >
               Movie Tracker 
            </h1>,
            <h3>
               Login 
            </h3>,
            null,
            <input
              name="email"
              onChange={[Function]}
              placeholder="Email address"
              type="text"
              value=""
            />,
            <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />,
            <button
              className="login-button"
              onClick={[Function]}
              type="submit"
            >
              Login
            </button>,
            <button
              className="create-user"
              onClick={[Function]}
            >
              Create User
            </button>,
            null,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="home-link"
              to="/"
            >
              Home
            </NavLink>,
          ],
          "className": "Login-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Movie Tracker ",
              "className": "Login-header",
            },
            "ref": null,
            "rendered": " Movie Tracker ",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Login ",
            },
            "ref": null,
            "rendered": " Login ",
            "type": "h3",
          },
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email address",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
              "className": "login-button",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Login",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create User",
              "className": "create-user",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Create User",
            "type": "button",
          },
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Home",
              "className": "home-link",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
